src/kixi/collect/process_manager/collection_request.clj

```diff
@@ -0,0 +1,281 @@
+                ::sharing-reject/original
+                ::ms/id]))
+
+;;
+
+(defmethod c/event-payload
+  [:kixi.datastore/sharing-changed "1.0.0"]
+  [_]
+  ::ms/sharing-change-payload)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Events -> Commands
+
+(defmethod c/event-type->command-types
+  [:kixi.collect/collection-requested "1.0.0"]
+  [_]
+  #{[:kixi.datastore/sharing-change "2.0.0"]})
+
+(defmethod c/event-type->command-types
+  [:kixi.datastore/sharing-changed "1.0.0"]
+  [_]
+  #{[:kixi.mailer/send-group-mail "1.0.0"]})
+
+(defmethod c/event-type->command-types
+  [:kixi.mailer/group-mail-accepted "1.0.0"]
+  [_]
+  #{[:kixi.collect.process-manager.collection-request/complete-process "1.0.0"]})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+
+(defn get-db-id
+  [m]
+  (get m :kixi.command/id))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Create Commands
+
+(defn create-update-sharing-permission-command
+  [state {:keys [::ms/id ::cr/group-collection-requests]} _]
+  [{:kixi.command/type :kixi.datastore/sharing-change
+    :kixi.command/version "2.0.0"
+    ::ms/id id
+    ::ms/sharing-update ::ms/sharing-conj
+    ::ms/activity ::ms/bundle-add
+    :kixi.group/id (first (first group-collection-requests))}
+   {:partition-key id}])
+
+(defn create-send-group-email-command
+  [state {:keys [:kixi.group/id]} _]
+  [{:kixi.command/type :kixi.mailer/send-group-mail
+    :kixi.command/version "1.0.0"
+    :kixi.mailer/source "support@mastodonc.com"
+    :kixi.mailer/destination {:kixi.mailer.destination/to-groups #{id}}
+    :kixi.mailer/message {:kixi.mailer.message/subject (str "You've been invited to contribute to a Datapack")
+                          :kixi.mailer.message/body
+                          {:kixi.mailer.message/text
+                           "<<&env.default_header>>This needs to be filled out.<<&env.default_footer>>"
+                           :kixi.mailer.message/html
+                           "<<&env.default_header>>This needs to be filled out.<<&env.default_footer>>"}}}
+   {:partition-key id}])
+
+(defn complete-process
+  [state-machine-actions state event backend]
+  (let [bid (get-in state [:value ::batch-id])
+        this-cr-id (get-in state [:value ::cr/id])
+        batch-results (get-batch backend bid)
+        requests-in-batch (get-in state [:value ::cr/ids])
+        grpd-batch-results (->> (update (group-by ::cr/id batch-results) this-cr-id conj {::action "complete"})
+                                (vals)
+                                (map #(set (map (comp keyword ::action) %))))]
+    (when (every? (partial = state-machine-actions) grpd-batch-results)
+      (log/info (str "Collection Request process was completed (batch " bid ", " (count batch-results)")"))
+      [(merge {:kixi.command/type :kixi.collect.process-manager.collection-request/complete-process
```

Comment:
Add documentation re having events return events

Reply:
events should not return events